apps/interpret_nn/halide/common_halide.cpp

```diff
@@ -78,12 +79,10 @@ Expr round_shift_right(const Expr &x, const Expr &exponent) {
     Type t = x.type();
     Type t_unsigned = t.with_code(halide_type_uint);
     Expr uexponent = cast(t_unsigned, exponent);
-    // Exponent must satisfy 0 <= exponent <= 31
-    // TODO: Maybe this should be an offset added to x prior to shifting.
-    Expr mask = (cast(x.type(), 1) << uexponent) - 1;
-    Expr remainder = x & mask;
-    Expr threshold = (mask >> 1) + (x < 0);
-    return (x >> uexponent) + (remainder > threshold);
+    Expr one = cast(x.type(), 1);
+    // This is intended to pattern-match against rounding-shift-right instruction generation
```

Comment:
Add comment that it assumes the type is int32, otherwise the rounding add would need to be a saturating add.

Reply:
I hadn't realize that saturating was part of this. I'd better go back and look at the spec. (And will add the comment.)

Reply:
...or better yet, just make it a saturating add now :-)